src/OrchardCore/OrchardCore.ContentManagement/DatabaseContentDefinitionStore.cs

```diff
@@ -25,7 +25,7 @@ public Task<ContentDefinitionRecord> LoadContentDefinitionAsync()
         /// <summary>
         /// Gets a single document (or create a new one) for caching and that should not be updated.
         /// </summary>
-        public Task<ContentDefinitionRecord> GetContentDefinitionAsync()
+        public Task<(bool, ContentDefinitionRecord)> GetContentDefinitionAsync()
```

Comment:
I think that the "caching" information should not leak at this level. It should always be either cacheable or not. Another solution is to change this method to two:
- CreateContentDefinitionRecord
- GetContentDefinitionRecord

And the user would check if Get returned null before calling Create. Right now it's more like a GetOrCreate() and the cachability of the result will vary.

Reply:
Okay, i'm open to any suggestions

That said, this is no more the case in the `distributedCache` branch, it is done at the `IDocumentStore` level and only checked at the `IDocumentManager<>` level, then any service using an `IDocumentManager<>` don't have to care about this.

    /// <inheritdoc />
    public Task<ContentDefinitionRecord> LoadContentDefinitionAsync() => _documentManager.GetMutableAsync();

    /// <inheritdoc />
    public Task<ContentDefinitionRecord> GetContentDefinitionAsync() => _documentManager.GetImmutableAsync();

> It should always be either cacheable or not

Yes i agree but this is the singularity i described in my comments. In some rare cases the caller should not create a new instance but should not cache the returned one. But this only happens in a given scope where we can't refresh the cache for a document that has already been loaded for update and that may have been flushed.

> it's more like a GetOrCreate() and the cachability of the result will vary.

Yes exactly, before it was already a `GetOrCreate()` with a factory, but, because of the above, even the result is not null it doesn't mean that it can be cached

I will think about it